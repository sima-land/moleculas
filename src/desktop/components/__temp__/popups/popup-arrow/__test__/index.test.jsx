import React from 'react';
import { shallow } from 'enzyme';
import PopupArrow from '..';

let arrow;

describe('<PopupArrow />', () => {
  it('renders correctly', () => {
    arrow = shallow(<PopupArrow />);
    expect(arrow).toMatchSnapshot();
    arrow = shallow(<PopupArrow shadow color='dark-blue' direction='bottom' />);
    expect(arrow).toMatchSnapshot();
    arrow = shallow(<PopupArrow color='dark-blue' className='test-class' direction='top' />);
    expect(arrow).toMatchSnapshot();
  });

  it('has correct attributes', () => {
    arrow = shallow(<PopupArrow className='test' shadow position={{ top: '10px', left: '100px' }} />);
    expect(arrow.find('span').prop('style')).toEqual({ top: '10px', left: '100px' });
  });
});
