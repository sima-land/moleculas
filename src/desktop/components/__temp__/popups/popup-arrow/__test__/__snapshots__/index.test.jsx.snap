// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PopupArrow /> renders correctly 1`] = `
<span
  className="base white top shadow-top"
  style={Object {}}
/>
`;

exports[`<PopupArrow /> renders correctly 2`] = `
<span
  className="base dark-blue bottom shadow shadow-bottom"
  style={Object {}}
/>
`;

exports[`<PopupArrow /> renders correctly 3`] = `
<span
  className="base dark-blue test-class top shadow-top"
  style={Object {}}
/>
`;
